src/proto/streams/prioritize.rs

```diff
@@ -736,6 +765,8 @@ impl Prioritize {
                             tracing::trace_span!("updating stream flow").in_scope(|| {
                                 stream.send_flow.send_data(len);
 
+                                self.current_send_buffer_size -= len as usize;
```

Comment:
I think this area needs to potentially wake up any tasks that have been waiting on "capacity", since now there is "more buffer space to use". The reason it needs to be done here, is because with flow control capacity, the connection doesn't actually get more until the peer has sent another WINDOW_UPDATE frame. So `recv_window_update` will notify a task there. But buffer size isn't affected by WINDOW_UPDATE frames.

Reply:
I'm not sure I understand. Why do we even call `self.flow.assign_capacity(len);` below that line, if we are actually supposed to wait for the peer to send another WINDOW_UPDATE frame?

Reply:
I am even more consumed that we call `self.flow.send_data(len);` immediately afterwards we called `self.flow.assign_capacity(len);`.

Reply:
Doesn't that mean that `self.flow.available` never changes from `Prioritize::pop_frame`? I don't understand the logic there.

Reply:
I called `assign_connection_capacity` from `poll_complete` after the `reclaim_frame` call.